src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Kafka/KafkaHeadersCollectionAdapter.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Datadog.Trace.Headers;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Kafka
+{
+    internal struct KafkaHeadersCollectionAdapter : IHeadersCollection
+    {
+        private static readonly IDatadogLogger Logger = DatadogLogging.GetLoggerFor<KafkaHeadersCollectionAdapter>();
+        private readonly IHeaders _headers;
+
+        public KafkaHeadersCollectionAdapter(IHeaders headers)
+        {
+            _headers = headers;
+        }
+
+        public IEnumerable<string> GetValues(string name)
+        {
+            // This only returns the _last_ bytes. Accessing other values is more expensive and should generally be unneccessary
+            if (_headers.TryGetLastBytes(name, out var bytes))
```

Comment:
How many bytes does this read? Are we sure we never read a partial multi-byte character?

Reply:
It reads all the bytes for the last instance of the header with name `name`. Headers are stored as raw bytes in Kafka, but you can have multiple headers with the same name, hence "last" in the name.

Reply:
Oh, so it's "get the bytes for the last header," not "get the last {count} bytes". Thanks for clarifying!